cdap-data-fabric/src/main/java/co/cask/cdap/data2/datafabric/dataset/service/executor/InMemoryDatasetOpExecutor.java

```diff
@@ -61,6 +61,9 @@ public DatasetSpecification create(Id.DatasetInstance datasetInstanceId, Dataset
 
     // TODO: Note - for now, just sending the name. However, type likely needs to be namesapce-aware too.
     DatasetSpecification spec = type.configure(datasetInstanceId.getId(), props);
+    if (props.getDescription() != null) {
+      spec = spec.setDescription(props.getDescription());
+    }
```

Comment:
setting originalProperties is done inside the type.configure(). Not that one or the other is better, but I would like it better if patching the description is done in the same place as patching the properties. Would you mind moving this into DatasetType.configure()?